src/YesSql.Core/Sql/BaseComandInterpreter.cs

```diff
@@ -246,23 +246,27 @@ public virtual IEnumerable<string> Run(IDropForeignKeyCommand command)
 
         private void Run(StringBuilder builder, ICreateColumnCommand command)
         {
+            var ignoreCollation = false;
             // name
             builder.Append(_dialect.QuoteForColumnName(command.ColumnName)).Append(Space);
 
             if (!command.IsIdentity || _dialect.HasDataTypeInIdentityColumn)
             {
                 builder.Append(_dialect.GetTypeName(command.DbType, command.Length, command.Precision, command.Scale));
-            }
-
-            if (_dialect.ColumnCollation != string.Empty)
-            {
-                builder.Append(Space).Append(_dialect.ColumnCollation);
+                ignoreCollation = false;
             }
 
             // append identity if handled
             if (command.IsIdentity && _dialect.SupportsIdentityColumns)
             {
                 builder.Append(Space).Append(_dialect.IdentityColumnString);
+                ignoreCollation = true;
+            }
+
+            // collation
+            if (!ignoreCollation)
```

Comment:
I knew this is weird little bit, because you inject the `Integer` data type on the `IdentityColumnString`, IMHO we should rid off the data type from `IdentityColumnString`, so this below line will be applied for all the columns in the table
```csharp
builder.Append(_dialect.GetTypeName(command.DbType, command.Length, command.Precision, command.Scale));
```